app/controllers/lti_nrps_controller.rb

```diff
@@ -0,0 +1,132 @@
+class LtiNrpsController < ApplicationController
+  respond_to :json
+  skip_before_action :set_course
+  skip_before_action :authorize_user_for_course
+  skip_before_action :authenticate_for_action
+  skip_before_action :update_persistent_announcements
+
+  rescue_from LtiLaunchController::LtiError, with: :respond_with_lti_error
+  def respond_with_lti_error(error)
+    Rails.logger.send(:warn) { "Lti NRPS Error: #{error.message}" }
+    render json: { error: error.message }.to_json, status: :bad_request
+  end
+  action_auth_level :request_access_token, :instructor
+  def request_access_token
+    # get private key from config to sign Autolab's client assertion as a JWK
+    private_key = Rails.configuration.lti_settings["tool_private_key"].to_s.gsub(/\\n/, "\n")
+    tool_rsa_private = OpenSSL::PKey::RSA.new(private_key)
+    # build JWK format for RSA
+    optional_parameters = {
+      kid: "dGkeZwrt+H7GnGZ2t2LUfhYf+/o=",
+      use: 'sig',
+      alg: 'RS256',
+      e: "AQAB",
+      kty: "RSA",
+    }
+    tool_rsa_private_JWK = JWT::JWK.new(tool_rsa_private, optional_parameters);
```

Comment:
Nit: redundant semicolon. 

There are several more in `lti_launch_controller.rb`

Reply:
fixed